Code/Unsupervised_reconstruction/Source/reconstructionModel.py

```diff
@@ -9,14 +9,14 @@
 
 import matplotlib.pyplot as plt
 
-from models.types import (ModernLanguages, Operations, InferenceData,
+from models.types import (ModernLanguages, Operations, InferenceData_Samples,
                           InferenceData_Cognates, InferenceData_SamplesEmbeddings, 
                           PADDING_TOKEN)
 
 
-from source.editModel import EditModel
-from source.packingEmbedding import PackingEmbedding
-from source.dynamicPrograms import compute_mutation_prob, compute_posteriors
+from Source.editModel import EditModel
+from Source.packingEmbedding import PackingEmbedding
+from Source.dynamicPrograms import compute_mutation_prob, compute_posteriors
 from models.probcache import ProbCache
 
 device = 'cuda' if torch.cuda.is_available() else 'cpu'
```

Comment:
For the one that I remember, there are:
- Source/dynamicPrograms.py
- Source/editModel.py
- data/vocab.py (all the `ComputeInferenceData_*` method returns the input for the reconstruction model and then the device should be defined there)
- as a consequence of the input tensors' device computed in the file above, the prior language models should also have a device related to `ReconstructionModel`. A solution may be to incorporate the prior language model in the `ReconstructionModel` instances so that the device can be internally set up.